src/ReinforcementLearningCore/src/policies/agent/agent_base.jl

```diff
@@ -13,22 +13,21 @@ is a Callable and its call method accepts varargs and keyword arguments to be
 passed to the policy. 
 
 """
-mutable struct Agent{P,T,C} <: AbstractPolicy
+mutable struct Agent{P,T} <: AbstractPolicy
     policy::P
     trajectory::T
-    cache::C # need cache to collect elements as trajectory does not support partial inserting
 
     function Agent(policy::P, trajectory::T) where {P,T}
-        agent = new{P,T, SRT}(policy, trajectory, SRT())
+        agent = new{P,T}(policy, trajectory)
 
         if TrajectoryStyle(trajectory) === AsyncTrajectoryStyle()
             bind(trajectory, @spawn(optimise!(policy, trajectory)))
         end
         agent
     end
 
-    function Agent(policy::P, trajectory::T, cache::C) where {P,T,C}
-        agent = new{P,T,C}(policy, trajectory, cache)
+    function Agent(policy::P, trajectory::T) where {P,T,C}
```

Comment:
C is not needed any more? Or why did you keep it here?

Reply:
It's an oversight. Already fixed.